src/redismodule.h

```diff
@@ -955,11 +955,11 @@ typedef struct RedisModuleTypeMethods {
 #define REDISMODULE_ATTR REDISMODULE_ATTR_COMMON
 #endif
 
-REDISMODULE_API void * (*RedisModule_Alloc)(size_t bytes) REDISMODULE_ATTR;
-REDISMODULE_API void * (*RedisModule_TryAlloc)(size_t bytes) REDISMODULE_ATTR;
-REDISMODULE_API void * (*RedisModule_Realloc)(void *ptr, size_t bytes) REDISMODULE_ATTR;
+REDISMODULE_API __attribute__((alloc_size(1))) void * (*RedisModule_Alloc)(size_t bytes) REDISMODULE_ATTR;
```

Comment:
Done with adding REDISMODULE_ATTR_ALLOC in https://github.com/redis/redis/pull/11982/commits/5a7c0153a017bac23cfeaf17d0918bcd7c91edbf#diff-26821aebe5bf67b83e43e39c63ad877ff6caa7b023e88712d5ab28463974542a

Reply:
Sorry I probably don't understand this change fully, but it looks like it exposes the allocator interface in modules as being standard allocators using `alloc_size`, is that correct?  If yes, then this may end up exposing the extra space usage problem to users of modules too, i.e. user applications may request `X` bytes and `RedisModule_MallocUsableSize` may return `X+n` bytes available, thus resulting in the breakage externally.